faker/providers/company/it_IT/__init__.py

```diff
@@ -330,6 +330,24 @@ class Provider(CompanyProvider):
 
     company_suffixes = ('SPA', 'e figli', 'Group', 's.r.l.')
 
+    @staticmethod
+    def _luhn_checksum(number):
+        def digits_of(n):
+            return [int(d) for d in str(n)]
+        digits = digits_of(number)
+        odd_digits = digits[-1::-2]
+        even_digits = digits[-2::-2]
+        checksum = 0
+        checksum += sum(odd_digits)
+        for d in even_digits:
+            checksum += sum(digits_of(d * 2))
+        return checksum % 10
+
+    @staticmethod
+    def _calculate_luhn(partial_number):
```

Comment:
Since this logic gets reused a lot, we just added it in https://github.com/joke2k/faker/blob/master/faker/utils/checksums.py.

Could you refactor to use the functions from `utils`?

Reply:
Should be fine now